File: line/frontend/src/app/modules/manufacturingOrder/manufacturingOrderProduct/manufacturingOrderProduct.ts
import { Shared } from '../../../shared/shared';
import { Product } from '../../product/product';

export class ManufacturingOrderProduct {
    id: number = Shared.generateId();;
    width : number;
    height : number;
    quantity : number;
    product : Product = new Product();

    constructor(options?: any){
    	this.id = (options && options.id) ? options.id : Shared.generateId();
    	this.width = (options && options.width) ? options.width : undefined;
    	this.height = (options && options.height) ? options.height : undefined;
    	this.quantity = (options && options.quantity) ? options.quantity : undefined;
    	this.product = (options && options.product) ? options.product : new Product();
    }
}

File: line/line/src/main/java/com/todotresde/siglo21/line/model/ManufacturingOrderProduct.java
package com.todotresde.siglo21.line.model;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

/**
 * Created by Leonardo on 26/12/2016.
 */
@Entity
@Table(name = "manufacturingOrderProduct")
public class ManufacturingOrderProduct {
    @Id
    @NotNull
    private Long id;
    private Double width;
    private Double height;
    @NotNull
    private Integer quantity;
    @NotNull
    private Long product;

    public ManufacturingOrderProduct(){

    }

    public ManufacturingOrderProduct(Long id, Double width, Double height, Integer quantity, Long product) {
        this.id = id;
        this.width = width;
        this.height = height;
        this.quantity = quantity;
        this.product = product;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Double getWidth() {
        return (width != null) ? width : 0;
    }

    public void setWidth(Double width) {
        this.width = width;
    }

    public Double getHeight() {
        return (height != null) ? height : 0;
    }

    public void setHeight(Double height) {
        this.height = height;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public Long getProduct() {
        return product;
    }

    public void setProduct(Long product) {
        this.product = product;
    }

}
